lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -840,11 +840,10 @@ CStackBasicDescriptor CGHeroInstance::calculateNecromancy (const BattleResult &b
 				}
 				else
 				{
-					auto quality = [getCreatureID](std::shared_ptr<Bonus> pick) -> std::vector<int>
+					auto quality = [getCreatureID](std::shared_ptr<Bonus> pick) -> std::tuple<int, int, int>
 					{
 						const CCreature * c = VLC->creh->objects[getCreatureID(pick)];
-						std::vector<int> v = {c->level, static_cast<int>(c->cost.marketValue()), -pick->additionalInfo[1]};
-						return v;
+						return std::tuple<int, int, int> {c->level, static_cast<int>(c->cost.marketValue()), -pick->additionalInfo[1]};
 					};
 					if(quality(topPick) < quality(newPick))
```

Comment:
I don't understand, how it works? and how does it work now?

Reply:
Later we compare two tuples, before it was 2 vectors

Reply:
But operator < is not defined for vectors or I don't know something?

Reply:
https://en.cppreference.com/w/cpp/container/vector/operator_cmp

Reply:
according to that, it's gone since c++20. It make sense to replace deprecated stuff (maybe not in this pr but still)

Reply:
as for this pr, std::array<int, 3> can be used instead of tuple since there is no type difference

Reply:
We can leave it as a std::tuple. https://stackoverflow.com/questions/48445291/stdtuple-faster-than-stdarray I based it off this which says std::tuple has better perf.

Also, std::vector still has comparison operators as of C++20. They're just implemented using the spaceship operator instead.

Reply:
As far as I understood it is not gone, just replaced, or changed signature. But yes no objection to replace vector with tuple. Btw is tuple size limited somehow? I mean can it contain 20 items?

Reply:
Tuple is not size-limited. It's implemented as a variadic template and the size is determined at compile-time so you can add as much stuff as you want.

For vector, the compare operation can still be performed, but the `operator<` has been replaced with `operator<=>`. More info on `<=>` here: https://devblogs.microsoft.com/cppblog/simplify-your-code-with-rocket-science-c20s-spaceship-operator/